test/integration/enumerate_databases.test.ts

```diff
@@ -0,0 +1,121 @@
+import { expect } from 'chai';
+
+import { AddUserOptions, MongoClient, MongoServerError } from '../../src';
+
+describe('listDatabases', function () {
+  let client: MongoClient;
+
+  beforeEach(async function () {
+    client = this.configuration.newClient();
+    await client.connect();
+  });
+
+  afterEach(async function () {
+    await client?.close();
+  });
+
+  it('should return an Array', async () => {
+    const dbInfo = await client.db().admin().listDatabases();
+    expect(Array.isArray(dbInfo.databases)).to.be.true;
+  });
+
+  it('should contain no duplicates', async () => {
```

Comment:
So, this setup doesn't really create any databases, right? If we really want to ensure we are testing the no duplicates assertion, I think we have to first create the conditions that could conceivably cause duplicates (e.g., try to create more than one database with the same name). Testing that an empty set is empty doesn't really accomplish that.

Reply:
Since it's just not possible to create a duplicate named DB I feel like that would be testing MongoDB more than our driver. The spec doesn't require us to return an array, it says we could return an iterable, I think the test's goal is to make sure that one's iterable doesn't mistakenly loop over a value twice. If the server somehow handed us back two DB's with the same name, currently our logic wouldn't do anything to stop that from reaching the user, which I think is "more" correct in that its passing along the server's mistake.  

This test and the  one above are described in the spec's test plan, so pulling them  in even though they're not really significant for our API feels right for completion sake.

Reply:
can we at least assert that the list is not empty?

Reply:
Also, if there are prose tests here, can we separate them into a prose test file and explicitly include the test plan instructions?

Reply:
Reorganized and added length assertions as well as a check that it results in an array of objects. I pulled in the test plan in a comment at the top of the new file